WalkingChallenge/DropDownPicker.swift

```diff
@@ -0,0 +1,195 @@
+
+import UIKit
+
+class DropDownPicker: UIControl {
+  fileprivate var picker: UIPickerView!
+  fileprivate var toolbar: UIToolbar!
+
+  var textfield: UITextField!
+
+  var data: Array<String>?
+
+  var placeholder: String = "Tap to choose ..." {
+    didSet { textfield.placeholder = placeholder }
+  }
+  var placeholderWhileSelecting: String? = "Pick an option ..."
+
+  var selectedIndex: Int {
+    get {
+      if let text = textfield.text {
+        return data?.index(of: text) ?? -1
+      }
+      return -1
+    }
+    set {
+      if newValue < 0 {
+        textfield.text = nil
+        textfield.placeholder = placeholder
+      } else {
+        pickerView(self.picker, didSelectRow: newValue, inComponent: 0)
+      }
+    }
+  }
+
+  fileprivate var hasValidSelection: Bool {
+    if let text = textfield.text {
+      return (data?.contains(text))!
+    }
+    return false
+  }
+
+  fileprivate var previousSelection: String? = nil
+  fileprivate var hasValidPreviousSelection: Bool {
+    if let previousSelection = previousSelection {
+      return (data?.contains(previousSelection))!
+    }
+    return false
+  }
+
+  public init(frame: CGRect, textfield: UITextField, data: Array<String>? = nil) {
+    super.init(frame: CGRect.null)
+
+    self.textfield = textfield
+    self.textfield.delegate = self
+    self.textfield.placeholder = placeholder
+    (self.textfield.value(forKey: "textInputTraits") as! NSObject)
```

Comment:
Hmm, I couldn't find the property.  I think we can do that as a follow-up.  That would definitely be nicer.